File: BasicRxJavaSample/app/src/main/java/com/example/android/observability/demoTest/ImitateUserDao.java
package com.example.android.observability.demoTest;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import io.reactivex.Completable;
import io.reactivex.Flowable;

@Dao
public interface ImitateUserDao {
    @Query("SELECT * FROM users LIMIT 1")
    Flowable<ImitateUser> getImitateUser();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insertImitateUser(ImitateUser user);

    @Query("DELETE FROM Users")
    void deleteAllImitateUser();
}

File: BasicRxJavaSampleKotlin/app/src/main/java/com/example/android/observability/imitate/persistance/ImitateUserDao.kt
package com.example.android.observability.imitate.persistance

import androidx.room.*
import com.google.android.material.circularreveal.CircularRevealHelper
import io.reactivex.Completable
import io.reactivex.Flowable

@Dao
interface ImitateUserDao {

    @Query("SELECT * FROM users LIMIT 1")
    fun getUser():Flowable<ImitateUser>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun updateOrInsertUser(user: ImitateUser):Completable

    @Query("DELETE  FROM users")
    fun delAllImitateUser()
}